### PR TITLE
Add support for skipping certain kinds of kubernetes configs

### PR DESCRIPTION
**Pull request by joakimnordling**
Custom resource definitions are at the current time not supported by kubeval. This causes issues for validation of these types and due to this there were already some workarounds for the sealed secret related things.

This PR adds support for defining your own types to skip and removed the old workarounds used for sealed secrets controller and secrets. A benefit with this approach (compared to totally ignoring the files) is that kubeval will still check that the yaml syntax is valid for these files, so it could help catching some syntax errors.

### CODE REVIEW

File: tasks.py [hunk 1]
```diff
@@ -235,7 +229,9 @@ def _should_ignore(path):
         if not _should_ignore(path)
     ]
 
-    run(["kubeval"] + kube_yamls)
+    skip_kinds = ",".join(devops.settings.KUBEVAL_SKIP_KINDS)
```
Comment (lietu) [inline tasks.py]: I imagine this does not work well if `devops.settings.KUBEVAL_SKIP_KINDS = []`

File: devops/lib/component.py [hunk 2]
```diff
@@ -76,9 +76,11 @@ def validate(self, ctx=None):
         if not ctx:
             return
 
+        skip_kinds = ",".join(KUBEVAL_SKIP_KINDS)
+
         for file in self.kube_configs:
             path = self.kube_configs[file]
-            result = run(["kubeval", path])
+            result = run(["kubeval", "--skip-kinds", skip_kinds, path])
```
Comment (lietu) [inline devops/lib/component.py]: I imagine this also does not work well if `KUBEVAL_SKIP_KINDS = []`

File: tasks.py (hunk 1 above)
Comment (joakimnordling) [reply tasks.py]: If you configure it like that, then `skip_kinds` will be an empty string. So the end result is more or less equal to running something like `kubeval --skip-kinds "" kube/02-kured.yaml` which works nicely. This is actually one of the benefits of the subprocess.run, that it takes care of quoting attributes properly. For more details please see https://docs.python.org/3/library/subprocess.html#frequently-used-arguments

File: devops/lib/component.py (hunk 2 above)
Comment (joakimnordling) [reply devops/lib/component.py]: See the reply to the other question.

**Pull request merged**